### ISSUE TITLE
False positive of box_default

### ISSUE THREAD

**Issue by BusyJay** [rust-lang/rust-clippy#9639]
### Summary

`Box::default()` will not work if the input is `Box<dyn trait>`.

### Lint Name

box_default

### Reproducer

I tried this code:

```rust
fn check(_: Box<dyn Send>) {}

fn main() {
    check(Box::new(u64::default())); // vs check(Box::default());
}
```

I saw this happen:

```
warning: `Box::new(_)` of default value
 --> src/main.rs:4:11
  |
4 |     check(Box::new(u64::default())); // vs check(Box::default());
  |           ^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: use `Box::default()` instead
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#box_default
  = note: `#[warn(clippy::box_default)]` on by default

warning: `playground` (bin "playground") generated 1 warning
```

I expected to see this happen:
Compile OK.

### Version

```text
0.1.66 (2022-10-12 c0983a9)
```

**Comment by llogiq** [comment]
This is a duplicate of #9621 which by the way already has a not yet merged solution in #9622 and in fact the current master should already have a suggestion to use `Box::<u64>::default()`.
